src/sidebar/components/ShareDialog/ImportAnnotations.tsx

```diff
@@ -68,36 +83,22 @@ function ImportAnnotations({
     setError(null);
     setSelectedUser(null);
     readExportFile(file)
-      .then(annotations => {
-        setAnnotations(annotations);
-
-        // Pre-select the current user in the list, if at least one of the
-        // annotations was authored by them.
-        const userMatch = annotations.some(ann => ann.user === currentUser);
-        setSelectedUser(userMatch ? currentUser : null);
-      })
+      .then(setAnnotations)
```

Comment:
Moving update of the selected user into the render means that when the annotation list is fetched, two renders will happen in succession rather than one. What was the motivation for the change?

Reply:
The fact that the selected item now cannot be inferred from the list of annotations now, but from the `userList` instead.

So first, annotations need to be set, which triggers a re-render. That will force memoized `userList` to be recomputed, re-rendering again.

After that, we can check if `userList` contains current user, and set it as selectedUser.

I know it now renders twice because of that, but simplifies the logic, as we now have a single type to care about. `userList` is an array of that type, and `selectedUser` is an item of the same type.

Reply:
> The fact that the selected item now cannot be inferred from the list of annotations, but from the userList instead.

I'm not sure what you mean by "inferred from the list of annotations". Previously the selected item was either a userid associated with an entry in the `userList` (though an index could have been used instead) or `null` to represent "All annotations" (export) or "no user selected" (import).

Reply:
> I'm not sure what you mean by "inferred from the list of annotations". Previously the selected item was either a userid associated with an entry in the userList (though an index could have been used instead) or null to represent "All annotations" (export) or "no user selected" (import).

What I meant is that `selectedUser` is no longer `string | null`, but the object itself that is selected and passed as `value` to `SelectNext`.

This simplifies other logics and makes it easier to use together with `SelectNext`, but has the drawback if this second render.

Reply:
Very good idea.